src/services/cronJobs/fillSnapshotBalances.ts

```diff
@@ -46,88 +49,122 @@ export async function addFillPowerSnapshotBalanceJobsToQueue() {
   let offset = 0;
   let isFinished = false;
 
-  const groupByBlockNumbers: {
-    [p: number]: {
+  const groupByTimestamp: {
+    [timestamp: number]: {
       userId: number;
       powerSnapshotId: number;
       walletAddress: string;
     }[];
   } = {};
+
```

Comment:
We must fetch the least indexed block timestamp here and store it in a variable. Then compare the snapshot time to this value and fetch snapshot balance if least indexed timestamp is greater or equal to snapshot time

Reply:
Done